### DISCUSSION TITLE
Under extrusion at the beginning of every line??

### DISCUSSION THREAD

**Discussion by Spinboost** [moonglow/FlashForge_Marlin#140] [Q&A]
Hello, I'm not sure if this is an issue with the firmware or something with my settings but here's the issue and I'm hoping someone can help me out with..

TLDR: when using the LA version of the firmware for my Dreamer, every line starts out under-extruded regardless of settings

A few months ago I bought a dreamer that was "BRAND NEW IN BOX", turns out it had about 200 hours on it and was probably sitting in a garage for a few years before I got it. I started with the original firmware and FlashPrint, got it working in Cura instead, found the amazing firmware here and have been running it since, run a ton of tests and read everything I can. Over time I've had several re-flashes to figure out exactly what I needed. I was fighting ringing issues for a long time and over the past few months I've installed a glass plate, all new linear bearings, belts, Noctua fans for the power supply and drivers, TMC 2209's for all steppers. I've inspected everything, replaced nozzles and PFTE tubes in the hotend, re-insulated the hotend. I've had this printer completely disassembled several times now, learning everything the hard way. Currently I'm using [Tracedgod's Cura profile](https://github.com/tracedgod/FlashForge-Dreamer-Cura-Definitions) with a slightly different start code

Using the dir_swap variant, everything works OK but I get some pretty heavy corners and also underextrusion visible from top surfaces, changing flow rate in Cura to 102% fixes the top surfaces but the corners are about the same.

I changed to dir_swap_la and have been messing with that, still have to keep the flow at 102%, it seems every test gives me a k-value at about 0.22 for either extruder, which seems really high for a direct drive, but that's what the tests show every time..

Regardless of what settings I use I always end up with underextrusion at the beginning of every line as long as I'm using the la enabled firmware. I've played around with every setting I can think of, retraction distance, speed, prime speed, extra prime amounts, all speed/junction deviation/acceleration settings. I've confirmed extruder e-step calibration, I've followed both the [TeachingTech calibration guide](https://teachingtechyt.github.io/calibration.html#intro) and the [Ellis Print Tuning Guide](https://ellis3dp.com/Print-Tuning-Guide/) but always find that the default settings that come with the this firmware are about as spot-on as you can get except for Linear Advance. I've also tried different filament colors and brands of PLA and ABS, all of them are kept sealed with desiccant when not printing.

To narrow it down I've reset the EEPROM with m502 then m500, reset all cura settings back to TracedGods profile, flashed dir_swap, did a mesh bed level, printed a cube, then flashed dir_swap_la and printed the same file without changing any settings in EEPROM, then again with only changing the k-value to 0.22, and one last time changing the k-value to 0. The underextrusion at the beginning of a line happens on every one printed with dir_swap_la. Attached are some photos showing what I'm talking about.

Sorry if this is all over the place and rambling, if anyone has any suggestions or anything to try I would really appreciate it!

![PXL_20230206_022542703](https://user-images.githubusercontent.com/121821525/216876243-ca009b70-1883-4976-9875-b49ada2676e5.jpg)
![PXL_20230206_022442078](https://user-images.githubusercontent.com/121821525/216876245-adf9a3e4-b131-4bff-9698-5c945d7e8ee7.jpg)
![PXL_20230206_022501679](https://user-images.githubusercontent.com/121821525/216876246-2c1a8a68-0fde-454e-a77a-ff20385e338d.jpg)
![PXL_20230206_021909964](https://user-images.githubusercontent.com/121821525/216876247-07bb4d0b-eeec-4416-be85-51660b44ec86.jpg)

**Comment by tracedgod** [comment]
@Spinboost, Thanks for all the details decribing your issue! Admittedly, I have not actually tried the Linear Advance builds on my printer. If I understand correctly, this only happens when you try to use the ``dir-swap-la`` build, correct?

Doing some quick looks on the Marlin issue page, I did find a [similar issue](https://github.com/MarlinFirmware/Marlin/issues/23265) as you describe here, which seems to suggest that the timings were adjusted in their ``bugfix-2.1.x`` branch when using the Linear Advance features. The changes most likely were not back-ported to ``2.0.9.x`` branch, similar situation as issue #126.

**Reply by Spinboost** [reply]
@tracedgod, Yes that's correct, `dir-swap-la` build is the only one that I've experienced this, since getting the printer I've used `v0.15.0` `base`,` dir`, `dir_swap` and `dir_swap_la`. and with `v0.15.1` I've only used `dir_swap` and `dir_swap_la`

Thank you for [linking that Marlin issue](https://github.com/MarlinFirmware/Marlin/issues/23265) about LA timing, it sounds like that would solve the issue, it's kind of surprising that it's a problem to begin with though.

A couple of things I tried yesterday;
After reading [this](https://github.com/MarlinFirmware/Marlin/pull/24533) I thought maybe it's an issue with the drivers- so I swapped out the extruder drivers back to the stock FF ones (and swapped the wiring around to stock too). no change

Thinking about the timing issue [you sent](https://github.com/MarlinFirmware/Marlin/issues/23265) and how the planner couldn't look forward, I thought maybe it's just because it starts a line so slow, so I upped the print acceleration from the firmware default of 200 up to 1000, at 1000 I got a LA K-value of about 0.08 (before it was 0.22)
The underextrusion still exists but it's definitely less noticeable. but the tradeoff is a bunch of ringing from whipping that heavy extruder around.

So I guess here's my options

1. live with underextrusion - having Cura randomize start/endpoints would probably hide a lot of it
2. live with bulging corners (without LA enabled firmware)
3. live with ringing
4. find a balance between ringing and underextrusion, maybe by removing an extruder to get the weight down
5. or yaknow, just lower my standards a bit. But tinkering with this stuff in pursuit of perfection is what makes it fun!

If you or anyone else has any suggestions or ideas I'm all ears, I'll gladly be a guinea pig to test stuff out

 
![PXL_20230207_223618224](https://user-images.githubusercontent.com/121821525/217386679-8d966be7-b12b-4757-935b-18e50c7d1383.jpg)

A roll of filament was ending on this cube, so I just kinda.. shoved the next color in behind the old one, worked pretty well actually
![PXL_20230207_001921709](https://user-images.githubusercontent.com/121821525/217386682-9b39b65d-1e6b-4600-8e8c-be0c1c8ff3b5.jpg)

**Reply by qwertychouskie** [reply]
Once 2.1.x is ported, perhaps you can compile with Input Shaping to get rid of the ringing ;)

**Reply by Spinboost** [reply]
I'm excited to try it!

**Reply by digital0785** [reply]
> Once 2.1.x is ported, perhaps you can compile with Input Shaping to get rid of the ringing ;)

cannnt waaaiiiiiitttttt

**Comment by Cargo190** [comment]
Just wanted to confirm, yes, that's exactly how it is for me.
To improve the print quality, I completely changed the extruder to BIQU H2, increased accelerations  X,Y and got 0.05 when calibrating LA. However, the bad extrusion after changing the layer became even stronger and more noticeable. The ringing and echoes were also very strong.
